src/cartservice/src/Startup.cs

```diff
@@ -27,20 +27,31 @@ public Startup(IConfiguration configuration)
         public void ConfigureServices(IServiceCollection services)
         {
             string redisAddress = Configuration["REDIS_ADDR"];
+            string spannerProjectId = Configuration["SPANNER_PROJECT"];
+            string spannerInstanceId = Configuration["SPANNER_INSTANCE"];
+
             if (!string.IsNullOrEmpty(redisAddress))
             {
                 services.AddStackExchangeRedisCache(options =>
                 {
                     options.Configuration = redisAddress;
                 });
+                services.AddSingleton<ICartStore, RedisCartStore>();
+            }
+            else if (!string.IsNullOrEmpty(spannerProjectId))
+            {
+                if (string.IsNullOrEmpty(spannerInstanceId))
```

Comment:
Actually, could we move this logic/console.write in the `SpannerCartStore.cs`? It will allow to have the hard-coded `onlineboutique` value close to the source of the Spanner implementation :)

Reply:
Oh, actually I see what you have this here, that's because that what was/is existing for the `redis` implementation.

Reply:
So disregard my comment, let's keep that like you just did, and let's see if we could refactor that for both implementation `spanner` and `redis`, not very important at this stage.

Reply:
Cleaned it up: 6ad6e8eeeba7d952a78dc98b771d245d1bd4aed0